Only set phone on Customer.io profile when member is SMS subscriber

aaronschachter opened the pull request:
### What's this PR do?

This pull request modifies our upsert of a member's Customer.io profile to only set their `phone` attribute when they are subscribed to SMS promotions.

### How should this be reviewed?

👀 

### Any background context you want to provide?

This work is the first step towards only storing mobile numbers in Customer.io for members that are subscribed to SMS promotions -- that way we don't need to lookup each Northstar user when sending a Gambit broadcast to safety check that we're not accidentally sending a broadcast to a member who's unsubscribed.

The next step is removing phone numbers from all existing Customer.io profiles that are not subscribed to SMS promotions. This will likely be handled by running a Chompy import or by modifying the [BackfillCustomerIo]( https://github.com/DoSomething/northstar/blob/master/app/Console/Commands/BackfillCustomerIo.php) command to optionally accept a CSV of user ids (there's no need to backfill all users -- just ones who have a mobile set, are unsubscribed, and haven't had their promotions muted per our February Customer.io retirement import)

### Relevant tickets

References [Pivotal #177168236](https://www.pivotaltracker.com/story/show/177168236).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [x] Added appropriate feature/unit tests.
- [ ] If new attributes were added to users, there is a corresponding PR to surface these in Aurora.
- [ ] If new attributes were added to users, then the data team already knows about these changes.

mendelB reviewed (approved):
👏

mendelB commented:
Curious -- why can't we just rely on the SMS status itself on the profile instead of needing to remove the phone numbers? Or is it that the less PII the better in c.io?

aaronschachter commented:
>  why can't we just rely on the SMS status itself on the profile instead of needing to remove the phone numbers

We do this currently -- but Gambit will still make a Northstar lookup for each broadcast request, just to make sure we didn't accidentally send to unsubscribes via the Customer.io filter (it's happened before, which is why we added the Northstar lookup).

The current Gambit broadcast endpoint expects a NS ID -- our end goal, once we've removed all mobile numbers for non-subscribes, is that the endpoint will instead accept a `mobile` field,. We can assume all requests with a `mobile` are indeed ones we should send, in order to remove the Northstar lookup (and simplify scaling/network requests). Does that make sense?

mendelB commented:
> > why can't we just rely on the SMS status itself on the profile instead of needing to remove the phone numbers
> 
> We do this currently -- but Gambit will still make a Northstar lookup for each broadcast request, just to make sure we didn't accidentally send to unsubscribes via the Customer.io filter (it's happened before, which is why we added the Northstar lookup).
> 
> The current Gambit broadcast endpoint expects a NS ID -- our end goal, once we've removed all mobile numbers for non-subscribes, is that the endpoint will instead accept a `mobile` field,. We can assume all requests with a `mobile` are indeed ones we should send, in order to remove the Northstar lookup (and simplify scaling/network requests). Does that make sense?

Aha! thanks for clarifying, that def makes sense. So moving the source of truth over to the mobile field itself for a more assured filter removing the need to confirm with Northstar since the subscription filter via c.io could be wonky.
